rasa_core/policies/form_policy.py

```diff
@@ -78,6 +78,28 @@ def recall(self,
         # modify the states
         return self._recall_states(self._modified_states(states))
 
+    def state_is_unhappy(self, tracker, domain):
+        # since it is assumed that training stories contain
+        # only unhappy paths, notify the form that
+        # it should not be validated if predicted by other policy
+        tracker_as_states = self.featurizer.prediction_states(
+            [tracker], domain)
+        states = tracker_as_states[0]
+
+        memorized_form = self.recall(states, tracker, domain)
+
+        state_is_unhappy = (
+            memorized_form is not None and
+            memorized_form == tracker.active_form.get('name')
+        )
+
+        if state_is_unhappy:
```

Comment:
doesn't it make sense to leave the debug statement where it was initally? to avoid doing the if twice?

Reply:
it makes, but then in the second place I'll need to recalculate states

Reply:
or pass the states instead of tracker, which makes using this function less convenient

Reply:
ok fair, in that case feel free to merge. Have you opened a PR against NLU master as well yet?

Reply:
@akelad that would be [this one](https://github.com/RasaHQ/rasa_nlu/pull/3140#issuecomment-477933594) i believe

Reply:
oh man 🙈 so hard to keep track of everything haha. @Ghostvv can you make sure the changes are identical on both because they don't seem to be